python/sglang/srt/utils.py

```diff
@@ -453,14 +453,18 @@ def kill_parent_process():
     os.kill(parent_process.pid, 9)
 
 
-def monkey_patch_vllm_p2p_access_check():
+def monkey_patch_vllm_p2p_access_check(gpu_id: int):
     """
     Monkey patch the slow p2p access check in vllm.
     NOTE: We assume the p2p access is always allowed, which can be wrong for some setups.
     """
-    import vllm.distributed.device_communicators.custom_all_reduce_utils as tgt
 
-    setattr(tgt, "gpu_p2p_access_check", lambda *arg, **kwargs: True)
+    # TODO: need a better check than just dev str name match
+    # compat: skip RTX 40 series as they do not have P2P feature and even checking for them may cause errors
+    device_name = torch.cuda.get_device_name(gpu_id)
+    if "RTX 40" not in device_name:
+        import vllm.distributed.device_communicators.custom_all_reduce_utils as tgt
+        setattr(tgt, "gpu_p2p_access_check", lambda *arg, **kwargs: True)
```

Comment:
should we directly return false here?

Reply:
@merrymercy  Tested using `setattr(tgt, "gpu_p2p_access_check", lambda *arg, **kwargs: False)` and the system locked up with 2x4090 tp=2.  Have no idea why.  So not doing the monkeypatch works but forcing it to return false also breaks.

Reply:
@Qubitium The system seems not to be locked up but keeps checking all the connections between any two peers like this

```
Original exception was:
Error in sys.excepthook:

Original exception was:
[rank0]:[W CudaIPCTypes.cpp:16] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]
Error in sys.excepthook:
```

And then it gets a cache.